src/Chat/OllamaChat.php

```diff
@@ -0,0 +1,270 @@
+                'prompt' => $prompt,
+                'stream' => true,
+            ]
+        );
+
+        return $this->decodeStreamOfText($response);
+    }
+
+    /**
+     * Send a chat request
+     *
+     * @see https://github.com/ollama/ollama/blob/main/docs/api.md#generate-a-chat-completion
+     *
+     * @param  Message[]  $messages
+     */
+    public function generateChat(array $messages): string
+    {
+        $response = $this->sendRequest(
+            'POST',
+            'chat',
+            [
+                'model' => $this->config->model,
+                'messages' => $this->prepareMessages($messages),
+                'stream' => false,
+            ]
+        );
+        $json = $this->decodeJson($response->getBody()->getContents());
+
+        return $json['message']['content'];
+    }
+
+    /** @param Message[] $messages */
+    public function generateChatStream(array $messages): StreamInterface
+    {
+        $response = $this->sendRequest(
+            'POST',
+            'chat',
+            [
+                'model' => $this->config->model,
+                'messages' => $this->prepareMessages($messages),
+                'stream' => true,
+            ]
+        );
+
+        return $this->decodeStreamOfChat($response);
+    }
+
+    public function setSystemMessage(string $message): void
+    {
+        $this->systemMessage = Message::system($message);
+    }
+
+    /** @param FunctionInfo[] $tools */
+    public function setTools(array $tools): void
+    {
+        throw new MissingFeatureExcetion('This feature is not supported');
+    }
+
+    public function addTool(FunctionInfo $functionInfo): void
+    {
+        throw new MissingFeatureExcetion('This feature is not supported');
+    }
+
+    /** @param FunctionInfo[] $functions */
+    public function setFunctions(array $functions): void
+    {
+        throw new MissingFeatureExcetion('This feature is not supported');
+    }
+
+    public function addFunction(FunctionInfo $functionInfo): void
+    {
+        throw new MissingFeatureExcetion('This feature is not supported');
+    }
+
+    /**
+     * Send the HTTP request to Ollama API endpoint
+     *
+     * @param  mixed[]  $json
+     *
+     * @see https://github.com/ollama/ollama/blob/main/docs/api.md
+     */
+    protected function sendRequest(string $method, string $path, array $json): ResponseInterface
+    {
+        $response = $this->client->request($method, $path, ['json' => $json]);
+        $status = $response->getStatusCode();
+        if ($status < 200 || $status >= 300) {
+            throw new HttpExcetion(sprintf(
+                'HTTP error from Ollama (%d): %s',
+                $status,
+                $response->getBody()->getContents()
+            ));
+        }
+
+        return $response;
+    }
+
+    /**
+     * Decode a JSON string into an array
+     *
+     * @return mixed[]
+     */
+    protected function decodeJson(string $json): array
```

Comment:
Should we move this one in an utils class?

Reply:
Ok, I created a LLPhant\Utility class.